src/dwarf_parser.cpp

```diff
@@ -181,7 +181,9 @@ SizedType Dwarf::get_stype(Dwarf_Die &type_die, bool resolve_structs) const
     }
     case DW_TAG_structure_type:
     case DW_TAG_union_type: {
-      std::string name = dwarf_diename(&type_die);
+      std::string name = dwarf_hasattr(&type_die, DW_AT_name)
+                             ? dwarf_diename(&type_die)
+                             : "<anonymous>";
```

Comment:
Hm yeah, that's unfortunate. Maybe we could generate a unique name with a version of dwarf_peel_type that records the links taken? For now I guess an error is the best we can do here.

Reply:
> Hm yeah, that's unfortunate. Maybe we could generate a unique name with a version of dwarf_peel_type that records the links taken?

Maybe it would be sufficient to generate *any* unique name. Since the type is anonymous, it shouldn't be used anywhere outside of the container type so it shouldn't matter what's the name under which it is stored. OTOH, I'm not sure how well our AST passes (semantic analyser, codegen, etc.) deal with anonymous structs.

> For now I guess an error is the best we can do here.

That's certainly a good quick solution but we should eventually come up with something better.

Reply:
Okay well, I will open an issue to improve the anon struct/union handling and close this PR. Clearly this is not a real fix and something else entirely is needed.